### DISCUSSION TITLE
How to work with Records in fp-ts

### DISCUSSION THREAD

**Discussion by adinvadim** [gcanti/fp-ts#1661] [Q&A]
How can I work with Records in fp-ts. Want to assign other records, remove keys etc, buy always has an error, that I cant have optional properties in Record
```
Property 'a' is optional in type 'R' but required in type 'Record<keyof R, string | number | undefined>
```
Example
```typescript
interface A {
  a?: string;
  b: string;
  c: number;
}

const fn = (r: A) => pipe(
  r,
  R.deleteAt('c'),
)

```

This code throw type error
```
Property 'a' is optional in type 'R' but required in type 'Record<keyof R, string | number | undefined>
```

<img width="1078" alt="Снимок экрана 2022-01-16 в 15 15 49" src="https://user-images.githubusercontent.com/8646071/149659518-91169b9a-8892-42ef-af9a-b16ab4c4354e.png">

**Comment by mlegenhausen** [comment]
Types with a fixed index signature like an `interface` are not records. A record has for each possible key always the same value type. Your interface `A` is called a `struct` in `fp-ts` terms. It is recommended not to use any record functions on these types. If you want to remove a key from a struct you should define a function that transforms between the two types.

```ts
interface A {
  a?: string;
  b: string;
  c: number;
}

interface B extends Omit<A, 'a'> {}

function fn({ a, ...rest }: A): B {
    return rest
}
```
